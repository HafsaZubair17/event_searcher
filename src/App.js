import React, { useRef, useState } from "react";
import Header from "./Components/Views/HeaderView";
import Carousels from "./Components/Views/CarouselView";
import Artist from "./Components/Views/ArtistView";
import EventDisplay from "./Components/Views/Event/EventDisplayView";
import Spinner from "react-bootstrap/Spinner";

// Begin Main App
function App() {
  const eventLocation = useRef(null);
  const artistLocation = useRef(null);
  const [artistData, setArtistData] = useState("");
  const [initial, setInitial] = useState(false);
  const [artistName, setArtistName] = useState("");
  const [eventInitial, setEventInitial] = useState(false);
  const [eventData, setEventData] = useState("");
  const [loading, setLoading] = useState(false);
  const [eventLoading, setEventLoading] = useState(false);
  const eventLocate = () => {
    eventLocation.current.scrollIntoView({ behavior: "smooth" });
  };
  const artistLocate = () => {
    artistLocation.current.scrollIntoView({ behavior: "smooth" });
  };

  return (
    <div className="App">
      <header>
        {/* Header Component of the page */}
        <Header
          artistLocate={artistLocate}
          setArtistData={setArtistData}
          setInitial={setInitial}
          setEventInitial={setEventInitial}
          setArtistName={setArtistName}
          setLoading={setLoading}
        />
        {/* Carousel Under the Header */}
        <Carousels />
        {initial ? (
          !loading ? (
            <>
              <hr></hr>

              {/* Component to display the artist */}
              <Artist
                eventLocate={eventLocate}
                artistData={artistData}
                artistName={artistName}
                setEventInitial={setEventInitial}
                setInitial={setInitial}
                setEventData={setEventData}
                setEventLoading={setEventLoading}
              />
              <div ref={artistLocation}></div>
            </>
          ) : (
            <>
              <div className="d-flex justify-content-center">
                <Spinner className="spinner" animation="grow" variant="dark" />
              </div>
            </>
          )
        ) : (
          <>
            <div ref={artistLocation}></div>
          </>
        )}

        {eventInitial ? (
          !eventLoading ? (
            <>
              <div ref={eventLocation}></div>
              <hr></hr>
              {/* Component to display the Events for the artist */}
              <EventDisplay
                id="event"
                eventData={eventData}
                setEventInitial={setEventInitial}
                setInitial={setInitial}
              />
            </>
          ) : (
            <>
              <div className="d-flex justify-content-center">
                <Spinner className="spinner" animation="grow" variant="dark" />
              </div>
            </>
          )
        ) : (
          <div ref={eventLocation}></div>
        )}
      </header>
    </div>
  );
}

export default App;
