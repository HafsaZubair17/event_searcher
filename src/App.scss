@import "~bootstrap/scss/bootstrap";

// Theme Colors
$maincolor: #d8e3e7;
$altcolor: #126e82;

body {
  background-color: #132c33;
  font-family: Big Caslon, Book Antiqua, Palatino Linotype, Georgia, serif;
}
.d-flex {
  margin-top: 5rem !important;
}
.spinner-grow {
  height: 3rem !important;
  width: 3rem !important;
  color: $maincolor !important;
}
