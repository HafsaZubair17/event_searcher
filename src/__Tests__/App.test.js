import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

// App component testing
it("should take a snapshot of app", () => {
  const { asFragment } = render(<App />);

  expect(asFragment(<App />)).toMatchSnapshot();
});
