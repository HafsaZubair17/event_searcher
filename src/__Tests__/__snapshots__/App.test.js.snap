// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of app 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <div
        class="row"
      >
        <a
          href="https://www.bandsintown.com/"
          target="_blank"
        >
          <img
            alt="logo"
            class="logo"
            href="https://www.bandsintown.com/"
            src="/logo.png"
          />
        </a>
        <div
          class="MuiPaper-root ForwardRef-root-1 search MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="ForwardRef-searchContainer-7"
          >
            <div
              class="MuiInputBase-root MuiInput-root ForwardRef-input-6 MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused"
              data-testid="search"
            >
              <input
                class="MuiInputBase-input MuiInput-input"
                placeholder="Search the artist here!"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-searchIconButton-4"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root ForwardRef-icon-5"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root ForwardRef-iconButton-2 ForwardRef-iconButtonHidden-3"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root ForwardRef-icon-5"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <a
          aria-label="twitter"
          class="social-icon twitter"
          href="https://twitter.com/Bandsintown?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor"
          style="display: inline-block; width: 50px; height: 50px; position: relative; overflow: hidden; vertical-align: middle;"
          target="_blank"
        >
          <div
            class="social-container"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
          >
            <svg
              class="social-svg"
              style="border-radius: 50%; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; fill-rule: evenodd;"
              viewBox="0 0 64 64"
            >
              <g
                class="social-svg-background"
                style="transition: fill 170ms ease-in-out; fill: transparent;"
              >
                <circle
                  cx="32"
                  cy="32"
                  r="31"
                />
              </g>
              <g
                class="social-svg-icon"
                style="transition: fill 170ms ease-in-out; fill: #d8e3e7;"
              >
                <path
                  d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
                />
              </g>
              <g
                class="social-svg-mask"
                style="transition: fill 170ms ease-in-out; fill: #126e82;"
              >
                <path
                  d="M0,0v64h64V0H0z M44.7,25.5c0,0.3,0,0.6,0,0.8C44.7,35,38.1,45,26.1,45c-3.7,0-7.2-1.1-10.1-2.9 c0.5,0.1,1,0.1,1.6,0.1c3.1,0,5.9-1,8.2-2.8c-2.9-0.1-5.3-2-6.1-4.6c0.4,0.1,0.8,0.1,1.2,0.1c0.6,0,1.2-0.1,1.7-0.2 c-3-0.6-5.3-3.3-5.3-6.4c0,0,0-0.1,0-0.1c0.9,0.5,1.9,0.8,3,0.8c-1.8-1.2-2.9-3.2-2.9-5.5c0-1.2,0.3-2.3,0.9-3.3 c3.2,4,8.1,6.6,13.5,6.9c-0.1-0.5-0.2-1-0.2-1.5c0-3.6,2.9-6.6,6.6-6.6c1.9,0,3.6,0.8,4.8,2.1c1.5-0.3,2.9-0.8,4.2-1.6 c-0.5,1.5-1.5,2.8-2.9,3.6c1.3-0.2,2.6-0.5,3.8-1C47.1,23.4,46,24.5,44.7,25.5z"
                />
              </g>
            </svg>
          </div>
        </a>
      </div>
      <div
        class="carousel-container"
      >
        <div
          class="carousel slide carousel-fade"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li />
            <li />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <img
                alt="First slide"
                class="d-block w-100"
                src="/carousel1.jpg"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Are you ready to party?
                </h3>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Second slide"
                class="d-block w-100"
                src="/carousel2.jpg"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Missing you favourite artists?
                </h3>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Third slide"
                class="d-block w-100"
                src="/carousel3.jpg"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Search your favourite arstist's event now!
                </h3>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div />
      <div />
    </header>
  </div>
</DocumentFragment>
`;
